[ling zhi wei] Duke Increments

lzw12345 opened the pull request:
creating a pr as requested

j-lum committed:
Add support for Gradle workflow

j-lum committed:
Add sample checkstyle configuration

Add toolVersion block in to Gradle code sample to prevent errors.

j-lum committed:
Change file mode on `gradle` to be executable

j-lum committed:
Merge pull request #12 from j-lum/gradle+x

Change file mode on `gradle` to be executable (#9)

j-lum committed:
Add configuration for console applications

Gradle defaults to an empty stdin which results in runtime exceptions
when attempting to read from `System.in`. Let's add some sensible
defaults for students who may still need to work with the standard
input stream.

j-lum committed:
Merge pull request #13 from j-lum/javaexec

Add configuration for console applications

lzw12345 committed:
Merge remote-tracking branch 'origin/gradle'

lzw12345 committed:
Added gradle along with some additional plugins as recomeneded by the tutorial

lzw12345 committed:
implemented level-1

lzw12345 committed:
added functionality required for level-2

lzw12345 committed:
test

lzw12345 committed:
Revert "test"

This reverts commit 35ad5d87ce5c0e68d0ca53433382e830e8cb22d2.

lzw12345 committed:
testing branching

lzw12345 committed:
Revert "testing branching"

This reverts commit 72097c1a3a96fdca1d30abd520334392ed52598a.

lzw12345 committed:
testing branching

lzw12345 committed:
testing branching

lzw12345 committed:
testing branching

lzw12345 committed:
Revert "testing branching"

This reverts commit 0321b4512ed11f6a804d418fb73e3e6deab364c0.

lzw12345 committed:
Revert "testing branching"

This reverts commit d135be393eca3fdca7aec4be0a1fcb94f42a115a.

lzw12345 committed:
Added mark as done function

lzw12345 committed:
level-3

lzw12345 committed:
Added functionality for level 4

lzw12345 committed:
level-4

lzw12345 committed:
level-5

lzw12345 committed:
Basic error handling implemented

lzw12345 committed:
level-6

lzw12345 committed:
level-6

lzw12345 committed:
level-6

lzw12345 committed:
level-6

lzw12345 committed:
Revert "level-6"

This reverts commit 440966feb9739656e91f0f52a3f185bd957c5e09.

lzw12345 committed:
Revert "level-6"

This reverts commit 440966feb9739656e91f0f52a3f185bd957c5e09.

lzw12345 committed:
level-6

lzw12345 committed:
level-6

lzw12345 committed:
Merge branch 'master' of https://github.com/lzw12345/duke

lzw12345 committed:
Merge branch 'master' of https://github.com/lzw12345/duke

lzw12345 committed:
in process of adding parsing class

lzw12345 committed:
updating parser class

lzw12345 committed:
added enum and updated parser and storage classes

lzw12345 committed:
saving work

lzw12345 committed:
finished parser and storage class

lzw12345 committed:
added some error handling to storage class

lzw12345 committed:
starting level 8

lzw12345 committed:
adding java.time support

lzw12345 committed:
updated adding java.time classes

lzw12345 committed:
completed date time implementation

lzw12345 committed:
Merge tag 'Level-8'

# Conflicts:
#	tasklist.txt

lzw12345 committed:
no message

EugeneTeu commented on src/main/java/Duke.java:
```diff
@@ -1,10 +1,67 @@
                 + "| |_| | |_| |   <  __/\n"
                 + "|____/ \\__,_|_|\\_\\___|\n";
-        System.out.println("Hello from\n" + logo);
+        System.out.println(logo + "    ____________________________________________________________\n" +
```
DukeException class is not used. Might want to look into creating an abstraction for that

EugeneTeu commented on src/main/java/Task.java:
```diff
@@ -0,0 +1,25 @@
+public abstract class Task {
+    protected String Description;
+    protected boolean isDone;
+    protected DateTimeFormatter OUTPUT_FORMAT = DateTimeFormatter.ofPattern("dd MMMM hhmm a");
```
good use of DateTimeFormatter !

EugeneTeu commented on src/main/java/Parser.java:
```diff
@@ -0,0 +1,93 @@
+    private String description;
+    private String date = "";
+    private LocalDateTime date;
+    private boolean isSafe = true;
```
a more descriptive variable name could be used. Quite unclear what this represents at first glance

EugeneTeu commented on src/main/java/Parser.java:
```diff
@@ -0,0 +1,93 @@
+    private boolean isSafe = true;
+
+    public Parser(String fullCommand) {
+        Pattern command_format = Pattern.compile("(?<commandWord>\\w+)"
```
good use of regex wow!

EugeneTeu reviewed (commented):
General lack of javadocs. Good use of Regex! I liked the way you use the DateTimeFormatter, I believe i can learn from it!

lzw12345 committed:
creating packages

lzw12345 committed:
implemented packaging

lzw12345 committed:
finished adding text ui

joloong commented:
### Good work! 👍

Good use of Regex. It is something that I could adopt for my project.

One improvement I could suggest is that the commit messages could be written in imperative and first letter could be capitalized. E.g. "Add text ui" instead of "finished adding text ui".

lzw12345 committed:
Add JUnit tests

lzw12345 committed:
Added javadoc comments

lzw12345 committed:
Added find function

lzw12345 committed:
Merge branch 'branch-A-Javadoc'

# Conflicts:
#	src/main/java/Ui/TextUi.java
#	tasklist.txt

lzw12345 committed:
Added some commentss

lzw12345 committed:
Merge branch 'branch-A-CodingStandard'

# Conflicts:
#	src/main/java/Ui/TextUi.java
#	src/main/java/parser/Parser.java
#	src/main/java/tasklist/Task.java
#	src/main/java/tasklist/TaskList.java

lzw12345 committed:
Adding level 1-3 of gui

lzw12345 committed:
Adding more gui features

lzw12345 committed:
Saving work

lzw12345 committed:
Implemented basic GUI

lzw12345 committed:
Trying a new GUI

lzw12345 committed:
saving work

lzw12345 committed:
Setting up TableView

lzw12345 committed:
basic Level-10 completed

lzw12345 committed:
Implemented some Assertions

lzw12345 committed:
need to commit tasklist

lzw12345 committed:
Basic TableView completed

lzw12345 committed:
Seperated TableView into another class

lzw12345 committed:
Adding Travis

lzw12345 committed:
Scene Builder Keeps Crashing

lzw12345 committed:
completed cashflow class

lzw12345 committed:
Adding TableView for Cashflow

lzw12345 committed:
Setting up changing Table

lzw12345 committed:
Merge branch 'AltGUI' of https://github.com/lzw12345/duke into AltGUI

# Conflicts:
#	src/main/java/parser/Parser.java
#	src/main/java/tasklist/Task.java

lzw12345 committed:
save

lzw12345 committed:
Completed Json storage transition

lzw12345 committed:
Testing PR

lzw12345 committed:
For PR

lzw12345 committed:
Merge branch 'AltGUI'

# Conflicts:
#	src/main/java/Main.java
#	tasklist.txt

lzw12345 committed:
Added some javadoc comments and improved coding style

lzw12345 committed:
Merge pull request #5 from lzw12345/A-CodingStyle

Added some javadoc comments and improved coding style

lzw12345 committed:
Fixed Jar File

lzw12345 committed:
Major changes across files
-Added functionality for find notes and task
-Added more chatbot responses
-Added more exceptions

lzw12345 committed:
Set theme jekyll-theme-midnight

lzw12345 committed:
-Added Readme v0.1
-Added Textwrap in columns
-Added Darkmode to the UI

lzw12345 committed:
-Updated readme with images
-Added textwrapping on the source and category columns of noteview

lzw12345 committed:
Fixed broken image link

lzw12345 committed:
fixing broken image link again

lzw12345 committed:
-Created jar file for TaskBook 0.2
-Edited code to better match coding standards

lzw12345 committed:
Fixed broken readme link

lzw12345 committed:
Fixed image link again

lzw12345 committed:
-Added command summary to readme
-Added help command
-Changed the initial label

lzw12345 committed:
-Fixing image link to class website
